eng/Build.props

```diff
@@ -8,4 +8,9 @@
   <ItemGroup Condition="'$(DotNetBuildFromSource)' == 'true' or '$(DotNetBuild)' == 'true'">
     <ProjectToBuild Include="$(RepoRoot)src\Microsoft.NET.Sdk.Aspire\Microsoft.NET.Sdk.Aspire.csproj" />
   </ItemGroup>
+
+  <!-- When building from source, we want to use the live repo contents as opposed to cloning the repo. -->
+  <PropertyGroup Condition="'$(ArcadeBuildFromSource)' == 'true'">
+    <CopySrcInsteadOfClone>true</CopySrcInsteadOfClone>
```

Comment:
@MichaelSimons Source build was failing due to the fact that we were not specifying the PackageProjectUrl and it seems to be a requirement (not sure why just bumping the SDK caused this to fail). I added the property in the root Directory.Build.props but noticed build would still fail as it seems that source build leg would clone the aspire repo as opposed to using the changes in the changeset. For that, I've workaround it by setting this property. Can you confirm that this is correct?

Reply:
cc @dotnet/source-build-internal

Reply:
This change should not be necessary.  Did you see this problem locally or in CI?

Reply:
I did see a problem locally. Turns out that it is now a requirement to set PackageProjectUrl, which we weren't doing before. I added a change which does it, but because that change is being added by this PR, source build is still failing as it is cloning aspire (without that change) to build and that is failing. When I set this property it started working as it would just use the changeset which includes the setting of PackageProjectUrl.

Reply:
I just pushed a change to undo this, and we'll see if source build breaks.

Reply:
Build failed, looks like this property was required after all.

Reply:
Is it because source-build hasn't been updated to a new sha of the dotnet/aspire repo?

Reply:
I wanted to look at the binlog but because the build was cancelled it was not captured.

> Is it because source-build hasn't been updated to a new sha of the dotnet/aspire repo?

This is the repo level source-build so it is building with the commits in this PR.

Reply:
The inner clone should be cloning from the latest commit in this PR.

Reply:
More internal folks are reporting being blocked because they need the rest of the changes in this PR (updating to 8.0.200) to be able to build the repo using the new VS. @MichaelSimons I hope it's okay to go ahead and merge this in for now, and I'm more than happy to revert this specific change once we know what is going on and how to fix it.

Reply:
Yes, don't let this block your PR.  I opened a [draft PR](https://github.com/dotnet/aspire/pull/2236) so that I can get a binlog.  I can't reproduce this locally.